lib/engine/game/g_18_fl.rb

```diff
@@ -111,6 +111,66 @@ def hotel_value
         @phase.status.include?('hotels_doubled') ? 20 : 10
       end
 
+      # How far is the start_hex from one of the corporation's stations?
+      # This is by track, not as the crow flies.
+      # start_hex is either a City or a Town Part. Cannot be Jacksonville
+      #
+      # NOTE: 18FL does not have forks in plain track, only junctions
+      # Also because of this, we only need to return the hexes visited
+      #  and can use the exits on tiles, except....
+      # The disjoint cities in Jacksonville are a problem though
+      def distance_to_station(corporation, start_hex)
+        raise GameError, 'Cannot calculate token cost of Jax' if start_hex.id == 'H23'
+
+        goal_hexes = corporation.tokens.select(&:city).map { |t| [t.city.hex, t.city] }
+        # # If there is a goal token in non-connected (gray) jax, then we need special logic.
+        # special_jax = jax_token && hex_by_id('H23').tile.color != :gray
+        distance = 0
+        visited_hexes = []
+        start_hexes = [[start_hex, start_hex.tile.cities.first]]
+        until start_hexes.empty?
```

Comment:
how difficult is it to create a generic hex distance calculator in graph? i feel like you should be using graph here

Reply:
It would require a bit more thought and effort; this one works in the case that only track is lawson track. Actually, after some testing it doesn't even work properly on Jacksonville's disconnected cities and I haven't found an obvious reason why yet either

Reply:
I was able to fix the logic in this

Reply:
can we put this in graph?

Reply:
I removed explicit references to 18FL, added comments, and moved it to graph